resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -368,6 +368,25 @@ class YarnClusterSuite extends BaseYarnClusterSuite {
     )
     checkResult(finalState, result, "true")
   }
+
+  def createEmptyIvySettingsFile: File = {
+    val emptyIvySettings = File.createTempFile("ivy", ".xml")
+    Files.write("<ivysettings />", emptyIvySettings, StandardCharsets.UTF_8)
+    emptyIvySettings
+  }
+
+  test("SPARK-34472: non-local ivySettings file should be localized on driver in cluster mode") {
```

Comment:
Can we also test `clientMode = true` and assert that the file name hasn't been modified?

Could also be good to add tests for explicitly using `file:///` URI and for an expected failure when using `local:///` URI?

Reply:
Thanks for the suggestion! Caught an issue with handling of `file:///`. Added tests for `local` and `hdfs` schemes.